src/cargo/ops/cargo_install.rs

```diff
@@ -442,9 +442,11 @@ impl<'gctx> InstallablePackage<'gctx> {
             let src = staging_dir.path().join(bin);
             let dst = dst.join(bin);
             self.gctx.shell().status("Installing", dst.display())?;
-            fs::rename(&src, &dst).with_context(|| {
-                format!("failed to move `{}` to `{}`", src.display(), dst.display())
-            })?;
+            if !dry_run {
+                fs::rename(&src, &dst).with_context(|| {
+                    format!("failed to move `{}` to `{}`", src.display(), dst.display())
+                })?;
+            }
             installed.bins.push(dst);
             successful_bins.insert(bin.to_string());
```

Comment:
Should we include `installed.bins` and `successful_bins` pushes to only be on real runs?

Reply:
If i add `installed.bins` and `successful_bins` inside the if statement it change nothing in the output, according to you should i do this to avoid any potential side effect ?

Reply:
Oh yes indeed, the `InstallTracker` persist data on disk, so yes i think we should include them in the `if` statement.

Reply:
I made `installed.bins` and `successful_bins` pushes exclusive to real runs